src/features/historySlice.js

```diff
@@ -0,0 +1,108 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import {
+  getHistory,
+  addToHistory,
+  removeFromHistory,
+  removeAllHistory,
+} from "../services/services";
+
+const initialState = {
+  status: "idle",
+  error: null,
+  history: [],
+};
+
+export const loadHistory = createAsyncThunk(
+  "likes/loadHistory",
+  async (_, { rejectWithValue }) => {
+    return await getHistory(_, rejectWithValue);
+  }
+);
+
+export const postToHistory = createAsyncThunk(
+  "likes/postToHistory",
+  async (video, { rejectWithValue }) => {
+    return await addToHistory(video, rejectWithValue);
+  }
+);
+
+export const deleteFromHistory = createAsyncThunk(
+  "likes/deleteFromHistory",
+  async (videoId, { rejectWithValue, dispatch }) => {
+    return await removeFromHistory(videoId, rejectWithValue, dispatch);
+  }
+);
```

Comment:
This format of writing Async Thunks is very new for me!! Definitely one of the learning points from this pull request.

Reply:
Same here